Remove another JustBehave reference

AnthonySteele opened the pull request:
_Summarise the changes this Pull Request makes._

_Please include a reference to a GitHub issue if appropriate._

martincostello commented:
Given these PRs are/will be basically all the same, why not just do one PR with a commit per removal to make it just one PR to review instead of 16?

AnthonySteele commented:
OK, leave this open and I will add to it.

AnthonySteele commented:
I have removed `JustBehave` from all the tests in the `JustSaying.UnitTests.AwsTools.MessageHandling.Sqs` folder. This forms a natural unit of work, so it might be worth review and merge before moving on.

This is a unit of work (and was quick to do) because I added a simple base class `WhenPublishingTestBase` which takes the place of `XAsyncBehaviourTest<SqsPublisher>` and supplies the basic mechanism that these tests use.

So let's see what we think of that before moving on and attempting similar in the remaining `10` references to `JustBehave` in unit tests.

martincostello commented on JustSaying.UnitTests/AwsTools/MessageHandling/Sqs/WhenPublishingTestBase.cs:
```diff
@@ -0,0 +1,39 @@
+{
+    public abstract class WhenPublishingTestBase : IAsyncLifetime
+    {
+        protected SqsPublisher SystemUnderTest;
```
Make these properties, not fields.

AnthonySteele commented:
That is all of the `XAsyncBehaviourTest<T>` usages removed from unit tests.
There are still 9 usages of `XBehaviourTest<T>` in the unit tests.
I will look at these next.

AnthonySteele commented:
All clear in `JustSaying.UnitTests`.
There is no `JustBehave` reference to remove  from the unit tests, as it come in via `JustSaying.TestingFramework`, which both the nit tests and integration tests use.

martincostello commented on JustSaying.UnitTests/XBehaviourTest.cs:
```diff
@@ -0,0 +1,55 @@
+                SystemUnderTest = CreateSystemUnderTest();
+                When();
+            }
+            catch (Exception ex)
```
Could simplify this (I think) to:

```csharp
catch (Exception ex) when (_recordExceptions)
{
    ThrownException = ex;
}
```

AnthonySteele replied:
Ah the infamous `when` keyword, as avoided in this PR!

martincostello commented:
Unless you've already started, I'm just about to start porting the tests in the `WhenRegisteringHandlersViaResolver` folder in the integration tests.

AnthonySteele commented:
No, I haven't done anything with integration tests. Go ahead!

martincostello commented:
Changes made in #508.

AnthonySteele commented:
My preference is to approve and merge this one now, it's at a point where further work is a change of focus.
